text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,405 @@
+depend on `@deprecated` code, the implementation in `master` will be fixed to no longer defer into the `@deprecated`
+feature, and that fix will be forward-ported into the `next` branch in the exact same way that other changes are.
+
+### Resetting v1 feature flags
+
+When developing new behaviors that are incompatible with previous ones, feature flags were introduced to allow customers
+control of when they migrate over to the _new behavior_. However, when a new major version release is issued, the _new
+behavior_ ideally becomes the _only behavior_.
+
+In certain cases, users may be unable to migrate over to the _new behavior_ of a feature flag without incurring
+replacement of resources. Removing ability to toggle the feature off in a new major version would significantly
+complicate upgrading applications to the new major version. Feature flags will be critically evaluated, and in cases
+where a migration is impossible or difficult, the default state will be changed from disabled to enabled, but users will
+retain the ability to disable as needed.
+
+In order to reduce the codebase divergence between `master` and `next` (in order to minimize the amount of manual work
+involved in forward-porting features on `next`), existing feature flags must be made version-aware, such that they are
+enabled by default (and impossible to disable if a clean and reasonably simple migration is possible) when the current
+version is greater then `2.0.0-0`.
+
+The actual removal of code paths that support the _old behavior_ of feature flags that can no longer be disabled will be
+deferred to _after_ the new major version has reached _General Availability_.
+
+### Monolithic Packaging
+
+The AWS Construct Library will be released as a single artifact, instead of a collection of more than 100 libraries. The
+`monocdk-experiment` package has the tools necessary to re-package the existing _hypermodular_ libraries into a single
+packaging. Re-using that same process will allow keeping the difference between `master` and `next` as constrained as
+possible (reducing the need for manual intervention in forward-porting features).
+
+Consequently, the `monocdk-experiment` package will be renamed so that it is published and advertised as
+`@aws-cdk/stdlib` on the `master` branch. Availability of this library as part of CDK v1 is going to provide customers
+with a stepping stone to begin migrating their applications to CDK v2.
+
+Additionally, all modules that are re-packaged as part of `@aws-cdk/stdlib` will be marked as `"private": true` in the
+`next` branch, making these available _only_ via `@aws-cdk/stdlib` in CDK v2.
+
+## Documentation Updates
+
+Before formally announcing the availability of CDK v2 for early adopters, a documentation source must be availble to
+facilitate use of the new repository. The minimum bar should be availability of an API reference documentation for the
+v2 codebase that is usable.
+
```

Comment:
Just the lines below:
- Finalize breaking changes to be expected
- Have signed off on the v1 maintenance plan
- Have a (preliminary) migration guide ready

Reply:
What I meant to ask is what does API reference documentation support for 2.0 looks like? It would be useful to have a rough idea of the scope of work required. Things that come to mind: add support for multiple versions in the web UI, add support for submodules.

The reason I think this would be useful is to de-risk the project - if we know that this is going to be a major piece of work, we need to start tackling it sooner rather than later since it's in the critical path of the project.

Reply:
I support the notion of the API reference for CDK v2.0 have a similar layout to the current version, i.e. a separate page for each service/module. 

We've heard so much feedback that the non-TypeScript portions of the API reference are sub-optimal, we really need to address the pain points in v2. Most of the feedback is for the API ref boils down to:

1. the overview page is not the default landing page when you navigate to a module in the Python API re
2. not all of the code snippets translated from TypeScript to the other languages are 100% correct (debugging incorrect sample code in official AWS docs is a poor customer experience), 
3. there are no code snippets or meaningful README content for "cfn-only" modules, and some customers don't know how to use the L1 constructs available in these modules - leading them to think the module/service is not yet available in the CDK